spring-web/src/main/java/org/springframework/http/client/reactive/ClientHttpResponse.java

```diff
@@ -51,4 +57,24 @@ default String getId() {
 	 */
 	MultiValueMap<String, ResponseCookie> getCookies();
 
+	static long mergeMaxAgeAndExpires(@Nullable String maxAgeAttribute, @Nullable String expiresAttribute) {
+		if (maxAgeAttribute != null) {
+			return Long.parseLong(maxAgeAttribute);
+		}
+		else if (expiresAttribute != null) {
+			ZonedDateTime expiresDate = HttpHeaders.getZonedDateTime(expiresAttribute);
+
+			// Verify that the input date is in the future
+			ZonedDateTime now = ZonedDateTime.now(ZoneOffset.UTC);
+			if (expiresDate == null || expiresDate.isBefore(now)) {
+				return -1;
+			}
+			else {
+				// Calculate the difference in seconds
+				return ChronoUnit.SECONDS.between(now, expiresDate);
+			}
+		}
+		return -1;
```

Comment:
I believe it should be fine because that's the way it is done at all the places for which this fix is intended: [HttpComponentsClientHttpResponse](https://github.com/spring-projects/spring-framework/blob/e1cf2032951ca5956f3b7d77302bc01949e82fec/spring-web/src/main/java/org/springframework/http/client/reactive/HttpComponentsClientHttpResponse.java#L74),  [ReactorNetty2ClientHttpResponse](https://github.com/spring-projects/spring-framework/blob/e1cf2032951ca5956f3b7d77302bc01949e82fec/spring-web/src/main/java/org/springframework/http/client/reactive/ReactorNetty2ClientHttpResponse.java#L145)

Reply:
ok